R/ScpModelFit-Class.R

```diff
@@ -55,36 +52,31 @@ setClass("ScpModelFit", slots = c(
 ))
 
 ## Class constructors
-ScpModelFit <- function(n, p) {
-    stopifnot(n >= 0)
-    stopifnot(p >= 0)
-    stopifnot(length(n) == 1)
-    stopifnot(length(p) == 1)
+ScpModelFit <- function() {
     new(
-        "ScpModelFit", n = n, p = p, effects = List(),
+        "ScpModelFit", effects = List(),
         levels = List()
     )
 }
 
 ## ---- Getters ----
 
-scpModelFitN <- function(object) {
-    object@n
-}
 
-scpModelFitP <- function(object) {
-    object@p
+scpModelFitP <- function(object) { # changed
+    ## Should this throw an error if there are no coefficients?
```

Comment:
It's usually better practice to interact (eg asking questions or putting TODOs) using another way then code comments. For instance, you can put your question in a small text file and than paste it as a comment in the PR.

To answer the question: I find throwing an empty vector good enough (this should never happen anyway).